src/nupic/engine/YAMLUtils.cpp

```diff
@@ -54,39 +54,38 @@ static void _toScalar(const YAML::Node& node, boost::shared_ptr<Scalar>& s)
     NTA_THROW << "Internal error: attempting to convert YAML string to scalar of type Byte";
     break;
   case NTA_BasicType_UInt16:
-    node >> s->value.uint16;
+    s->value.uint16 = node.as<NTA_UInt16>(); 
```

Comment:
this is a low-level port of the existing logic, so that everything works with the new API. 
But it may not optimal. I think the `_toScalar` and `_toArray`  methods could be removed, replaced with YAML parsing. So could be `ntypes/Value.hpp` and this file `engine/YAMLUtils` with some refactoring.